src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -424,8 +441,13 @@ public Status launch(String appId) throws IOException {
     }
 
     public Status stop(String sessionId) throws IOException {
-        StandardResponse.Status status = sendStandard("urn:x-cast:com.google.cast.receiver", StandardRequest.stop(sessionId), DEFAULT_RECEIVER_ID);
-        return status == null ? null : status.status;
+        stopRequested = true;
+        try {
+            StandardResponse.Status status = sendStandard("urn:x-cast:com.google.cast.receiver", StandardRequest.stop(sessionId), DEFAULT_RECEIVER_ID);
+            return status == null ? null : status.status;
+        } finally {
+            stopRequested = false;
```

Comment:
It looks unreliable. From my understanding the message about closing will be received asynchronously. There is no guarantee that you will reach this code after you receive the 'close' response and thus you will most probably not process the close request correctly.